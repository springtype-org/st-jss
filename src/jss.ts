import { FontFaceProperties, KeyFrameProperties } from 'springtype-types';
import { API } from './interface/API';
import { ClassProperty } from './interface/ClassProperties';
import { makeStyles } from './makeStyles';
import { makeFont } from './makeFont';
import { makeAnimation } from './makeAnimation';

export const jss: API = {
  defaultUnit: 'px',
  classNamePrefix: 'st',
  instanceCounter: 0,
  makeAnimation: (animationName: string, style: Partial<KeyFrameProperties>) =>
    makeAnimation(animationName, style, jss),
  makeFont: (style: FontFaceProperties) => makeFont(style),
  makeStyles: <T extends string = string>(classProperty: ClassProperty<T>) => makeStyles<T>(classProperty, jss),
};
