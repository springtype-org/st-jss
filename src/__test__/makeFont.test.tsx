import { jss } from '../jss';

describe('makeFont', () => {
  it('check style output', () => {
    const style = jss.makeFont({
      // Normal jss
      fontFamily: 'Nonito',
      src: 'url(https://fonts.com/myfont)',
    });
    expect(style).toMatchSnapshot();
    const { children: childrenOfStyle, ...otherPropsOfStyle } = style;
    expect(childrenOfStyle[0]).toMatchSnapshot();
    expect(otherPropsOfStyle).toMatchSnapshot();
  });
});
