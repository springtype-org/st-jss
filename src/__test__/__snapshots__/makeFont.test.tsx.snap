// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeFont check style output 1`] = `
Object {
  "attributes": Object {
    "type": "text/css",
  },
  "children": Array [
    "@font-face {
font-family: Nonito
src: url(https://fonts.com/myfont)
}",
  ],
  "type": "style",
}
`;

exports[`makeFont check style output 2`] = `
"@font-face {
font-family: Nonito
src: url(https://fonts.com/myfont)
}"
`;

exports[`makeFont check style output 3`] = `
Object {
  "attributes": Object {
    "type": "text/css",
  },
  "type": "style",
}
`;
